pkg/syncer/types/commontypes.go

```diff
@@ -58,31 +59,38 @@ func InitConfigInfo(ctx context.Context) (*ConfigInfo, error) {
 // GetVirtualCenterInstance returns the vcenter object singleton.
 // It is thread safe.
 func GetVirtualCenterInstance(ctx context.Context, configTypes *ConfigInfo) (*cnsvsphere.VirtualCenter, error) {
-	onceForVirtualCenter.Do(func() {
-		log := logger.GetLogger(ctx)
-		var vcconfig *cnsvsphere.VirtualCenterConfig
-		vcconfig, vcerror := cnsvsphere.GetVirtualCenterConfig(ctx, configTypes.Cfg)
-		if vcerror != nil {
-			log.Errorf("failed to get VirtualCenterConfig. Err: %+v", vcerror)
-			return
-		}
+	if atomic.LoadUint32(&vCenterInitialized) == 0 {
```

Comment:
Is this step needed? Everything within this `if` scope is sufficient and correct..

Reply:
It won't be completely thread safe in that case. If we have two goroutines that gets past the first if condition and second check is not there, we might end up with two instances.

Reply:
Agreed we can do away with this step. But if the instance is already initialized, we will unnecessarily end up doing lock and unlock operations every time this function is invoked. So it's better to keep this check IMO.

Reply:
`atomic.LoadUint32` does a lock anyway :) Even without this, multiple threads wont be blocked for long because we lock, verify that `vCenterInitialized == 1` and unlock.

Reply:
Agreed we will achieve the same functionality that way also. 
But mutex lock and unlock is more expensive than atomic load AFAIK. So from a performance perspective, it will be better that we do an atomic load than mutex lock/unlock every time this function is invoked.